x/ccv/consumer/keeper/keeper.go

```diff
@@ -206,76 +207,79 @@ func (k Keeper) DeletePendingChanges(ctx sdk.Context) {
 	}
-	return maturingVSCPacket
+	return maturingVSCPackets
 }
 
-// GetAllPacketMaturityTimes returns a slice of all PacketMaturityTimes, sorted by vscIDs.
-func (k Keeper) GetAllPacketMaturityTimes(ctx sdk.Context) (maturingVSCPacket []consumertypes.MaturingVSCPacket) {
+// GetAllPacketMaturityTimes returns a slice of all PacketMaturityTimes, sorted by maturity times.
+//
+// Note that PacketMaturityTimes are stored under keys with the following format:
+// PacketMaturityTimeBytePrefix | maturityTime.UnixNano() | vscID
+// Thus, the returned array is in ascending order of maturityTimes.
+// If two entries have the same maturityTime, then they are ordered by vscID.
+func (k Keeper) GetAllPacketMaturityTimes(ctx sdk.Context) (maturingVSCPackets []consumertypes.MaturingVSCPacket) {
 	store := ctx.KVStore(k.storeKey)
 	iterator := sdk.KVStorePrefixIterator(store, []byte{types.PacketMaturityTimeBytePrefix})
 
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
-		vscId := binary.BigEndian.Uint64(iterator.Key()[1:])
-		maturityTime := binary.BigEndian.Uint64(iterator.Value())
+		var maturingVSCPacket consumertypes.MaturingVSCPacket
+		if err := maturingVSCPacket.Unmarshal(iterator.Value()); err != nil {
+			panic(fmt.Errorf("failed to unmarshal MaturingVSCPacket: %w", err))
+		}
 
-		maturingVSCPacket = append(maturingVSCPacket, consumertypes.MaturingVSCPacket{
-			VscId:        vscId,
-			MaturityTime: maturityTime,
-		})
+		maturingVSCPackets = append(maturingVSCPackets, maturingVSCPacket)
 	}
-	return maturingVSCPacket
+	return maturingVSCPackets
 }
 
 // SetPacketMaturityTime sets the maturity time for a given received VSC packet id
-func (k Keeper) SetPacketMaturityTime(ctx sdk.Context, vscId, maturityTime uint64) {
+func (k Keeper) SetPacketMaturityTime(ctx sdk.Context, vscId uint64, maturityTime time.Time) {
 	store := ctx.KVStore(k.storeKey)
-	timeBytes := make([]byte, 8)
-	binary.BigEndian.PutUint64(timeBytes, maturityTime)
-	store.Set(types.PacketMaturityTimeKey(vscId), timeBytes)
+	maturingVSCPacket := consumertypes.MaturingVSCPacket{
+		VscId:        vscId,
+		MaturityTime: maturityTime,
+	}
+	bz, err := maturingVSCPacket.Marshal()
+	if err != nil {
+		panic(fmt.Errorf("failed to marshal MaturingVSCPacket: %w", err))
+	}
+	store.Set(types.PacketMaturityTimeKey(vscId, maturityTime), bz)
 }
 
-// GetPacketMaturityTime gets the maturity time for a given received VSC packet id
-func (k Keeper) GetPacketMaturityTime(ctx sdk.Context, vscId uint64) uint64 {
+// PacketMaturityExists checks whether the packet maturity time for a given vscId and maturityTime exists
+func (k Keeper) PacketMaturityTimeExists(ctx sdk.Context, vscId uint64, maturityTime time.Time) bool {
 	store := ctx.KVStore(k.storeKey)
-	bz := store.Get(types.PacketMaturityTimeKey(vscId))
-	if bz == nil {
-		return 0
-	}
-	return binary.BigEndian.Uint64(bz)
+	bz := store.Get(types.PacketMaturityTimeKey(vscId, maturityTime))
+	return bz != nil
```

Comment:
Would an empty byte array still be considered valid here?

Reply:
As long as the state is not corrupted, `bz` should not be an empty byte array. Anyway, this method is only used in testing.